doc/en/changelog.rst

```diff
@@ -31,6 +31,263 @@ with advance notice in the **Deprecations** section of releases.
 
 .. towncrier release notes start
 
+pytest 8.3.0 (2024-07-20)
+=========================
+
+New features
+------------
+
+- `#12231 <https://github.com/pytest-dev/pytest/issues/12231>`_: Added `--xfail-tb` flag, which turns on traceback output for XFAIL results.
+
+  * If the `--xfail-tb` flag is not sent, tracebacks for XFAIL results are NOT shown.
+  * The style of traceback for XFAIL is set with `--tb`, and can be `auto|long|short|line|native|no`.
+  * Note: Even if you have `--xfail-tb` set, you won't see them if `--tb=no`.
+
+  Some history:
+
+  With pytest 8.0, `-rx` or `-ra` would not only turn on summary reports for xfail, but also report the tracebacks for xfail results. This caused issues with some projects that utilize xfail, but don't want to see all of the xfail tracebacks.
+
+  This change detaches xfail tracebacks from `-rx`, and now we turn on xfail tracebacks with `--xfail-tb`. With this, the default `-rx`/ `-ra` behavior is identical to pre-8.0 with respect to xfail tracebacks. While this is a behavior change, it brings default behavior back to pre-8.0.0 behavior, which ultimately was considered the better course of action.
+
+
+- `#12281 <https://github.com/pytest-dev/pytest/issues/12281>`_: Added support for keyword matching in marker expressions.
+
+  Now tests can be selected by marker keyword arguments.
+  Supported values are :class:`int`, (unescaped) :class:`str`, :class:`bool` & :data:`None`.
+
+  See :ref:`marker examples <marker_keyword_expression_example>` for more information.
+
+  -- by :user:`lovetheguitar`
+
+
+- `#12567 <https://github.com/pytest-dev/pytest/issues/12567>`_: Added ``--no-fold-skipped`` command line option
+
+  If this option is set, then skipped tests in short summary are no longer grouped
+  by reason but all tests are printed individually with correct nodeid in the same
+  way as other statuses.
+
+  -- by :user:`pbrezina`
+
+
+
+Improvements in existing functionality
+--------------------------------------
+
+- `#12467 <https://github.com/pytest-dev/pytest/issues/12467>`_: Migrated all internal type-annotations to the python3.10+ style by using the `annotations` future import.
+
+  -- by :user:`RonnyPfannschmidt`
```

Comment:
I suggest we move this to `Contributor-facing changes` as it is an internal change.